### PR TITLE
Tiff + JP2 ingest CLI

### PR DESCRIPTION
**Pull request by seanupton**
`tiff-ingest-cli` branch implements functionality described in #96 and #97, extending and leveraging previous PDF functionality developed for #95.

## Rake task changes
* There is now a single rake task for PDF, Image ingests:
```$ rake newspaper_works:ingest_issues -- --path=/path/to/your/pdf_or_tiff/batch```
* I have made an attempt to create task aliases (really, empty depending tasks) for backward-compatibility with existing documentation.

## Evaluation / testing

* This now has a dependency on `newspaper_works_fixtures` 0.3.1, which includes both JP2 and TIFF fixtures for ingest.
* Reviewers may use the rake task above to point at the respective fixtures path in that GEM to test ingest of sample content.

### CODE REVIEW

File: spec/lib/newspaper_works/ingest/image_ingest_issues_spec.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  let(:pub_path) { File.join(tiff_fixtures, lccn) }
+
+  let(:expected_paths) do
+    entries = Dir.entries(pub_path).map { |p| File.join(pub_path, p) }
+    entries.select { |p| File.directory?(p) && !File.basename(p).start_with?('.') }
+  end
+
+  describe " construction and metadata" do
+    it "constructs with path and publication" do
+      issues = described_class.new(pub_path, publication)
```
Comment (ebenenglish) [inline spec/lib/newspaper_works/ingest/image_ingest_issues_spec.rb]: This line is repeated in every spec, consider DRYing

File: spec/lib/newspaper_works/ingest/issue_images_spec.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+  # Publication for TIFF fixtures:
+  let(:publication) { NewspaperWorks::Ingest::PublicationInfo.new(lccn_tiff) }
+
+  # Publication for JP2 fixtures:
+  let(:publication_jp2) { NewspaperWorks::Ingest::PublicationInfo.new(lccn_jp2) }
+
+  describe "issue construction and metadata" do
+    it "constructs with path and publication" do
+      issue = described_class.new(tiff_issue_path, publication)
```
Comment (ebenenglish) [inline spec/lib/newspaper_works/ingest/issue_images_spec.rb]: this line repeated 3x, consider DRYing

File: lib/newspaper_works/ingest/image_ingest_issues.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        entries = Dir.entries(path).map { |n| File.join(path, n) }
+        entries.select { |p| !File.basename(p).start_with?('.') }.each do |p|
+          next unless File.directory?(p)
+          next unless path_validates?(p)
+          result.push(p)
+        end
+        @issue_paths = result
+      end
+
+      def info(path)
```
[the remark this answers is not in the document]
Comment (seanupton) [reply lib/newspaper_works/ingest/image_ingest_issues.rb]: This is invoked from a factory method in a mixin [here](https://github.com/marriott-library/newspaper_works/blob/tiff-ingest-cli/lib/newspaper_works/ingest/path_enumeration.rb#L16), for all the classes that are mapping-like objects where keys are paths, and values represent the metadata+data of file.  This mixin has just a [few expectations](https://github.com/marriott-library/newspaper_works/blob/tiff-ingest-cli/lib/newspaper_works/ingest/path_enumeration.rb#L3-L7) of its consuming classes.

File: lib/newspaper_works/ingest/issue_images.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        entries.sort.each do |p|
+          next unless File.ftype(p) == 'file'
+          ext = File.basename(p).downcase.split('.')[-1]
+          next unless IMAGE_EXT.include?(ext)
+          @pages.push(p)
+        end
+        @pages
+      end
+
+      def info(path)
```
[the remark this answers is not in the document]
Comment (seanupton) [reply lib/newspaper_works/ingest/issue_images.rb]: _See above discussion re: `PathEnumeration` mixin._

File: spec/lib/newspaper_works/ingest/image_ingest_issues_spec.rb (hunk 1 above)
Comment (seanupton) [reply spec/lib/newspaper_works/ingest/image_ingest_issues_spec.rb]: Agreed, will move this to a `let`.

File: spec/lib/newspaper_works/ingest/issue_images_spec.rb (hunk 2 above)
Comment (seanupton) [reply spec/lib/newspaper_works/ingest/issue_images_spec.rb]: Will also move this to a `let`.

**Pull request merged**